2-Pricing/src/components/CheckoutRow.tsx

```diff
@@ -0,0 +1,53 @@
+import { Cart } from "../App";
+import { Subscription } from "../constants";
+import { SubscriptionKey } from "./Cart";
+
+interface CheckoutRowProps {
+  isMonthlyView: boolean;
+  subscription: Subscription;
+  inCart: Cart;
+  setInCart: (cart: Cart) => void;
+}
+
+export const CheckoutRow = ({
+  isMonthlyView,
+  subscription,
+  inCart,
+  setInCart,
+}: CheckoutRowProps) => {
+  const key = subscription.name.toLowerCase() as SubscriptionKey;
+  const quantityInCart = inCart[key] ?? 0;
+
+  const handleIncrementCart = () => {
+    const newCart = { ...inCart };
+    newCart[key] = quantityInCart + 1;
+    setInCart(newCart);
```

Comment:
Grab `inCart` from the parameter of `setInCart`

Reply:
Do you mean this:
`setInCart((cart) => {
     cart[key] = quantityInCart + 1;
     return cart;
 })`
 
In this case, what would the `setInCart` signature be in the `CheckoutRowProps` interface?

Reply:
Yeah, that, but I think you should make the copy with the spread operator like you did before.

For the signature you would have to import the type of a react state setter, maybe [this](https://stackoverflow.com/questions/71297102/typescript-type-for-reacts-usestate-setter-when) can help.

Reply:
But if you follow the suggestion about changing the prop to `handleIncrementCart`, you won't have to declare it.